diff --git a/libs/core/async_sycl/include/hpx/async_sycl/sycl_future.hpp b/libs/core/async_sycl/include/hpx/async_sycl/sycl_future.hpp
index 8cc3d9d9b1..cd35aca679 100644
--- a/libs/core/async_sycl/include/hpx/async_sycl/sycl_future.hpp
+++ b/libs/core/async_sycl/include/hpx/async_sycl/sycl_future.hpp
@@ -24,6 +24,7 @@
 #include <hpx/modules/threading_base.hpp>
 #include <hpx/runtime_local/runtime_local_fwd.hpp>
 #include <hpx/runtime_local/thread_pool_helpers.hpp>
+#include <hpx/synchronization/once.hpp>
 
 #include <exception>
 #include <memory>
@@ -165,23 +166,8 @@ namespace hpx { namespace sycl { namespace experimental {
         // -------------------------------------------------------------
         /// Convenience wrapper to get future from just a queue
         /// Note: queue needs to be constructed with the in_order attribute
-        HPX_FORCEINLINE hpx::future<void> get_future(
-            cl::sycl::queue& command_queue)
-        {
-            HPX_ASSERT(queue.is_in_order());
-            return hpx::detail::try_catch_exception_ptr(
-                [&]() {
-                    // The SYCL standard does not include a eventRecord method Instead
-                    // we have to submit some dummy function and use the event the
-                    // launch returns
-                    cl::sycl::event event = command_queue.submit(
-                        [](cl::sycl::handler& h) { h.single_task([]() {}); });
-                    return get_future(event);
-                },
-                [&](std::exception_ptr&& ep) {
-                    return hpx::make_exceptional_future<void>(HPX_MOVE(ep));
-                });
-        }
+        hpx::future<void> get_future(
+            cl::sycl::queue& command_queue);
 #if !defined(__HIPSYCL__)
         /// Convenience wrapper to get future from just a queue using SYCL host tasks
         /// Note: queue needs to be constructed with the in_order attribute
diff --git a/libs/core/async_sycl/src/sycl_future.cpp b/libs/core/async_sycl/src/sycl_future.cpp
index b37f0b4f51..fd6c192a4a 100644
--- a/libs/core/async_sycl/src/sycl_future.cpp
+++ b/libs/core/async_sycl/src/sycl_future.cpp
@@ -10,13 +10,56 @@
 #include <hpx/allocator_support/internal_allocator.hpp>
 #include <hpx/assert.hpp>
 #include <hpx/async_sycl/sycl_future.hpp>
-#if !defined(HPX_COMPUTE_DEVICE_CODE)
 
 namespace hpx { namespace sycl { namespace experimental { namespace detail {
+  /// Utility function working around segfault on Intel GPU. Initializes something within the runtime by runnning
+  ///a dummy kernel
+  int touch_sycl_device_by_running_a_dummy_kernel(void) {
+      try {
+          cl::sycl::queue q(cl::sycl::default_selector_v, cl::sycl::property::queue::in_order{});
+          cl::sycl::event my_kernel_event = q.submit(
+              [&](cl::sycl::handler& h) {
+                  h.parallel_for(512, [=](auto i) {});
+              },
+              cl::sycl::detail::code_location{});
+          my_kernel_event.wait();
+          std::cout << "SYCL runtime has been initialized for HPX!" << std::endl;
+      } catch (cl::sycl::exception const& e) {
+          std::cerr << "(NON-FATAL) ERROR: Caught sycl::exception during HPX SYCL dummy kernel!\n";
+          std::cerr << " {what}: " << e.what() << "\n ";
+          std::cerr << "Continuing for now as error only occured in the dummy kernel meant to "
+		    << "initialize the device by first touch...\n";
+          return 2;
 
-    hpx::future<void> get_future(cl::sycl::event command_event)
-    {
-        return get_future(hpx::util::internal_allocator<>{}, command_event);
+      }
+      return 1;
+  }
+  /// Dummy variable to ensure the touch_sycl_device_by_running_a_dummy_kernel is being run
+  const int init_sycl_device_hydro =
+      touch_sycl_device_by_running_a_dummy_kernel();
+
+  hpx::future<void> get_future(cl::sycl::queue& command_queue)
+  {
+      HPX_ASSERT(queue.is_in_order());
+      return hpx::detail::try_catch_exception_ptr(
+          [&]() {
+              // The SYCL standard does not include a eventRecord method
+              // Instead we have to submit some dummy function and use the
+              // event the launch returns
+              cl::sycl::event event = command_queue.submit(
+                  [](cl::sycl::handler& h) { h.single_task([]() {}); });
+              return get_future(event);
+          },
+          [&](std::exception_ptr&& ep) {
+              return hpx::make_exceptional_future<void>(HPX_MOVE(ep));
+          });
+  }
+
+#if !defined(HPX_COMPUTE_DEVICE_CODE)
+
+  hpx::future<void> get_future(cl::sycl::event command_event)
+  {
+      return get_future(hpx::util::internal_allocator<>{}, command_event);
     }
 #if !defined(__HIPSYCL__)
     hpx::future<void> get_future_using_host_task(
@@ -26,5 +69,5 @@ namespace hpx { namespace sycl { namespace experimental { namespace detail {
             hpx::util::internal_allocator<>{}, command_event, command_queue);
     }
 #endif
-}}}}    // namespace hpx::sycl::experimental::detail
 #endif
+}}}}    // namespace hpx::sycl::experimental::detail
